src/components/CompanyFilterButton/test.tsx

```diff
@@ -83,6 +106,49 @@ describe('CompanyFilterButton', () => {
         await waitFor(() => expect(screen.queryByPlaceholderText('Search...')).toBeNull());
     });
 
+    test('selects company on "enter"', async () => {
+        const onChange = jest.fn();
+        renderWithClient(<CompanyFilterButton onChange={onChange} />, {
+            executeQuery: () =>
+                fromValue({
+                    data: {
+                        companies: [company],
+                    },
+                }),
+        });
+        const button = screen.getByText('By Company');
+        userEvent.click(button);
+        const input = await waitFor(() => screen.getByPlaceholderText('Search...'));
+        expect(screen.queryByText('Loading...')).toBeNull();
+        userEvent.type(input, 'tic');
+        fireEvent.keyDown(input, { key: 'Enter' });
+        expect(onChange).toHaveBeenCalledWith(expect.anything(), { value: company });
+        // Make sure the tooltip is hidden after the onchange
+        await waitFor(() => expect(screen.queryByPlaceholderText('Search...')).toBeNull());
+    });
+
+    test('navigates company by keyboard', async () => {
+        const onChange = jest.fn();
+        renderWithClient(<CompanyFilterButton onChange={onChange} />, {
+            executeQuery: () =>
+                fromValue({
+                    data: {
+                        companies: [company, companyTwo],
+                    },
+                }),
+        });
+        const button = screen.getByText('By Company');
+        userEvent.click(button);
+        const input = await waitFor(() => screen.getByPlaceholderText('Search...'));
+        expect(screen.queryByText('Loading...')).toBeNull();
+        userEvent.type(input, 'tic');
+        fireEvent.keyDown(input, { key: 'ArrowDown' });
+        fireEvent.keyDown(input, { key: 'Enter' });
+        expect(onChange).toHaveBeenCalledWith(expect.anything(), { value: companyTwo });
+        // Make sure the tooltip is hidden after the onchange
+        await waitFor(() => expect(screen.queryByPlaceholderText('Search...')).toBeNull());
+    });
```

Comment:
LOVE these tests! Esp the check at the end to verify that selecting actually closes it too.